test/Microsoft.AspNetCore.Proxy.Test/WebSocketsTest.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Proxy.Test
+{
+    public class WebSocketsTest
+    {
+        private static async Task<string> ReceiveTextMessage(WebSocket socket, int maxLen = 4096)
+        {
+            var buffer = new byte[maxLen];
+            var received = 0;
+            while (true)
+            {
+                var result = await socket.ReceiveAsync(new ArraySegment<byte>(buffer, received, maxLen - received), CancellationToken.None);
+                Assert.Equal(WebSocketMessageType.Text, result.MessageType);
+                received += result.Count;
+                if (result.EndOfMessage)
+                {
+                    return Encoding.UTF8.GetString(buffer, 0, received);
+                }
+                Assert.InRange(received, 0, maxLen);
+            }
+        }
+
+        [Fact]
+        public async Task ProxyWebSocketsSmokeTest()
```

Comment:
@Tratcher  @davidfowl  Shouldn't this have a skip condition for Win7 and Windows Server 2008?